src/plugins/hexnumber/hexnumber.h

```diff
@@ -0,0 +1,23 @@
+/**
+ * @file
+ *
+ * @brief Header for hexnumber plugin
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+
+#ifndef ELEKTRA_HEXNUMBER_H
+#define ELEKTRA_HEXNUMBER_H
+
+#include <kdbplugin.h>
+
+#define ELEKTRA_HEXNUMBER_META_TYPE "hexnumber"
```

Comment:
You have to separate meta data for internal purposes and for the user to be specified.

1. To remember what you have transformed to transform it back please use meta keys in `internal/<plugin>`
2. For what the user needs to specify, we should discuss this in #1398 first.

For non-internal meta-data please add the specification in doc/METADATA.ini and in the plugin's contract.